pkg/logs/client/tcp/destination.go

```diff
@@ -101,8 +101,8 @@ func (d *Destination) sendAndRetry(payload *message.Payload, output chan *messag
 
 		metrics.LogsSent.Add(1)
 		metrics.TlmLogsSent.Inc()
-		metrics.BytesSent.Add(int64(len(payload.Encoded)))
-		metrics.TlmBytesSent.Add(float64(len(payload.Encoded)))
+		metrics.BytesSent.Add(int64(payload.UnencodedSize))
```

Comment:
the `UnencodedSize` [is not set for TCP payloads](https://github.com/DataDog/datadog-agent/blob/main/pkg/logs/sender/stream_strategy.go#L37) as TCP uses the stream strategy which does no encoding. 

This is a bit of an implementation detail but in order for this to work you could either set the `UnencodedSize` to `len(payload.Encoded)` at the line I linked above - or revert this back to use `len(payload.Encoded)` here.

Reply:
Ah, thank you for catching - I've changed to set UnencodedSize explicitly, which I think makes the intent more obviously correct for the metrics.